.github/workflows/release.yml

```diff
@@ -105,4 +105,458 @@ jobs:
+          chmod +x cosign-linux-amd64
+          sudo mv cosign-linux-amd64 /usr/local/bin/cosign
+
+      - name: Check Installed Cosign Version
+        run: |
+
+          # Get installed and latest versions
+          installed_version=$(cosign version | awk '/GitVersion:/ {print $2}')
+          latest_version=$(curl -s https://api.github.com/repos/sigstore/cosign/releases/latest | jq -r '.tag_name')
+
+          # Compare versions and provide appropriate output
+          if [[ "$installed_version" = "$latest_version" ]]; then
+            echo "Cosign is up to date (version $installed_version)"
+          else
+            echo "Cosign is not up to date"
+            echo "Installed version: $installed_version"
+            echo "Latest version: $latest_version"
+          fi
+      
+      - name: Check if COSIGN_PASSWORD exists in GitHub secrets
+        run: |
+
+          if 
+            ! gh secret get COSIGN_PASSWORD > /dev/null 2>&1; then
+            RANDOM_PASSWORD=$(openssl rand -base64 32)
+            echo $RANDOM_PASSWORD | gh secret set COSIGN_PASSWORD -
+            echo "COSIGN_PASSWORD secret set."
+  
+            # Removing the secret from the local runner's environment
+            # No sensitive initialized secrets should be left in the environment
+            echo "Removing RANDOM_PASSWORD from local runner."
+            unset RANDOM_PASSWORD
+          fi  
+
+      # These key pairs should not be removed from the secrets once generated as the public key is used by the end user for verification.
+      # Frequent changes in these key pairs can make it easier for an insider to exploit vulnerabilities.
+      # Using key management services is the best possible approach to protect against insider attacks.
+      # KMS can harden the key management process and provide better security.
+      # Changes would be visible in more transparent and secure manner to both external and internal stakeholders and thus any insider attack can be detected.
+      # Cosign natively supports many key management services such as Google Cloud KMS, Azure Key Vault, AWS KMS, Hashicorp Vault, etc.
+      # Keyless signing can also be considered for future purposes.
+      # Currently it may not be possible to use keyless Cosign signing with GitHub action identity provider.
+      # Sigstore community is actively working on various such projects to provide native tools integration with many CI providers such as GitHub, GitLab, etc.
+      # We can explore the above given options in the future to further harden up our Supply chain security, but for time being let's just get up and running with the current solution. 
+      - name: Generating Cosign Key
+        id: cosign_key
+        run: |
+          if [[ -z "${{ secrets.COSIGN_PUBLIC_KEY }}" || -z "${{ secrets.COSIGN_PRIVATE_KEY }}" ]]; then
+            cosign generate-key-pair github://paralus/paralus
```

Comment:
Q: What happens to the key pairs generated using this command? Where do they get stored? How end users can retrieve the public key for verification?

Reply:
It will check if either one of the variables doesn't exist. If it doesn't then it will regenerate the key-value pair.

Reply:
Let's remove this. We can always use cosign keys from github secret for now. Later we can move to keyless  method or kms based key.